src/Symfony/Component/Clock/DatePoint.php

```diff
@@ -66,6 +66,30 @@ public static function createFromMutable(\DateTime $object): static
         return parent::createFromMutable($object);
     }
 
+    public static function createFromTimestamp(int|float $timestamp): static
+    {
+        if (\PHP_VERSION_ID >= 80400) {
+            return parent::createFromTimestamp($timestamp);
+        }
+
+        if (\is_int($timestamp)) {
+            return (new static())->setTimestamp($timestamp);
+        }
+
+        if (!is_finite($timestamp) || \PHP_INT_MAX + 1.0 <= $timestamp || \PHP_INT_MIN > $timestamp) {
+            throw new \DateRangeError(sprintf('DateTimeImmutable::createFromTimestamp(): Argument #1 ($timestamp) must be a finite number between %s and %s.999999, %s given', \PHP_INT_MIN, \PHP_INT_MAX, $timestamp));
+        }
+
+        $microsecond = (int) ((abs($timestamp) - (int) abs($timestamp)) * 1_000_000);
+
+        if (0 > $timestamp) {
+            $microsecond = 1_000_000 - $microsecond;
+        }
+
+        return (new static())->setTimestamp((int) floor($timestamp))
+            ->setMicrosecond($microsecond);
```

Comment:
```suggestion
        return static::createFromFormat('U.u', sprintf('%.6F', $timestamp));
```

Reply:
Your implementation is incorrect for negative floats. This PR contains a test that fails with your implementation.

Reply:
However, given it's faster, I'll use it for positive floats.